### PR TITLE
Update workflow for publishing to GitHub Pages.

### PR DESCRIPTION
**Pull request by sengi**
- Make the GitHub Pages custom domain match the repo name.
- Add workflow_dispatch so we can trigger it ad-hoc if need be.
- Update to current versions of actions.
- Fix YAML lints in the workflow definition.

### CODE REVIEW

**Review by kevindew — COMMENTED** [review]
Looks good to me just a minor thing and a question:

With us changing domains do we need to put a redirect on the old hostname or are we confident we can clean up the links? (I'm guessing just a find and replace on dev docs will have most of it done)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,6 +21,6 @@ jobs:
         uses: peaceiris/actions-gh-pages@v3
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./build
-          cname: govuk-k8s-user-docs.publishing.service.gov.uk
+          publish_dir: ./public
```
[the remark this answers is not in the document]
Comment (sengi) [reply .github/workflows/ci.yml:24]: Oh thanks, I wrongly assumed this was a self-contained thing. I guess it makes sense to leave the directory name alone then.

[the remark this answers is not in the document]
Comment (sengi) [reply .github/workflows/ci.yml]: Change it back so it matches again 👍

**Review by kevindew — APPROVED** [review]
Nice one 👍 

Ah no I was going to approve but caught the build / public thing just before

**Pull request merged**